src/sentry/utils/snowflake.py

```diff
@@ -18,35 +27,65 @@ def validate(self, value):
         return True
 
 
-SENTRY_EPOCH_START = datetime(2022, 4, 26, 0, 0).timestamp()
 ID_VALIDATOR = SnowflakeBitSegment(settings.SNOWFLAKE_ID_LENGTH, "Snowflake ID")
 
 VERSION_ID = SnowflakeBitSegment(settings.SNOWFLAKE_VERSION_ID_LENGTH, "Version ID")
 TIME_DIFFERENCE = SnowflakeBitSegment(settings.SNOWFLAKE_TIME_DIFFERENCE_LENGTH, "Time difference")
 REGION_ID = SnowflakeBitSegment(settings.SNOWFLAKE_REGION_ID_LENGTH, "Region ID")
 REGION_SEQUENCE = SnowflakeBitSegment(settings.SNOWFLAKE_REGION_SEQUENCE_LENGTH, "Region sequence")
 
+MAX_AVAILABLE_REGION_SEQUENCEs = 1 << REGION_SEQUENCE.length
+
+
+# might need call this for every timestamp when generating id -> depends on implementation
+def get_region_sequence_assignment_order():
+    values = list(range(MAX_AVAILABLE_REGION_SEQUENCEs))
+    r = random.Random(BASE_SEED ^ int(datetime.now().timestamp()))
```

Comment:
The timestamp should be passed in as an argument. We want the assignment order to be derived the same way for the same timestamp.

When we backtrack to a previous timestamp, this also means we should pass in that one to get a different assignment order.